### ISSUE TITLE
Runtime error using API.retrieve()

### ISSUE THREAD

**Issue by ZachSand** [JonathanVusich/pcpartpicker#44]
I wanted to test out this API so I used pip to install it and setup a simple test file similar to the example provided in the README.md.

When attempting
```
from pcpartpicker import API

api = API()
cpu_data = api.retrieve("cpu")
```

I get the following error:
```
Traceback (most recent call last):
  File "/Users/me/PycharmProjects/pcpartpicker-comparison/test.py", line 5, in <module>
    cpu_data = api.retrieve("cpu")
  File "/Users/me/PycharmProjects/pcpartpicker-comparison/venv/lib/python3.8/site-packages/pcpartpicker/api.py", line 55, in retrieve
    return self._handler.retrieve(*args, force_refresh=force_refresh)
  File "/Users/me/PycharmProjects/pcpartpicker-comparison/venv/lib/python3.8/site-packages/pcpartpicker/handler.py", line 90, in retrieve
    raw_data: Dict[str, str] = loop.run_until_complete(self.scraper.retrieve(parts_to_download))
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/asyncio/base_events.py", line 616, in run_until_complete
    return future.result()
  File "/Users/me/PycharmProjects/pcpartpicker-comparison/venv/lib/python3.8/site-packages/pcpartpicker/scraper.py", line 44, in retrieve
    raise result
  File "/Users/me/PycharmProjects/pcpartpicker-comparison/venv/lib/python3.8/site-packages/aiohttp/client.py", line 989, in send
    return self._coro.send(arg)
  File "/Users/me/PycharmProjects/pcpartpicker-comparison/venv/lib/python3.8/site-packages/aiohttp/client.py", line 473, in _request
    conn = await self._connector.connect(
  File "/Users/me/PycharmProjects/pcpartpicker-comparison/venv/lib/python3.8/site-packages/aiohttp/connector.py", line 522, in connect
    proto = await self._create_connection(req, traces, timeout)
  File "/Users/me/PycharmProjects/pcpartpicker-comparison/venv/lib/python3.8/site-packages/aiohttp/connector.py", line 853, in _create_connection
    _, proto = await self._create_direct_connection(
  File "/Users/me/PycharmProjects/pcpartpicker-comparison/venv/lib/python3.8/site-packages/aiohttp/connector.py", line 968, in _create_direct_connection
    transp, proto = await self._wrap_create_connection(
  File "/Users/me/PycharmProjects/pcpartpicker-comparison/venv/lib/python3.8/site-packages/aiohttp/connector.py", line 924, in _wrap_create_connection
    await self._loop.create_connection(*args, **kwargs))
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/asyncio/base_events.py", line 1010, in create_connection
    sock = await self._connect_sock(
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/asyncio/base_events.py", line 924, in _connect_sock
    await self.sock_connect(sock, address)
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/asyncio/selector_events.py", line 494, in sock_connect
    return await fut
RuntimeError: await wasn't used with future
```

I'm testing this in PyCharm on MacOS (10.14.6) using python 3.8. 

Is this API still usable? I noticed it attempts to use the cached data at https://jonathanvusich.github.io/pcpartpicker-scraper/ but I get a 404 when attempting to access that page.

**Comment by JonathanVusich** [comment]
The Github pages site does not have an `index.html` page so the 404 error is most likely expected. It looks like there is an issue with the `aiohttp` library that this API depends on. Does this work on Python 3.7?

**Comment by ZachSand** [comment]
Created a new venv with Python 3.7 and the same error occurred. 

I then attempted to upgrade the aiohttp library from 3.5.4 to 3.6.2 and dacite from 1.0.0 to 1.5.0 (the latest for the two) and both 3.7 and 3.8 still had the original error.

**Comment by JonathanVusich** [comment]
I'm afraid I can't help very much in this regard. I just checked the CI tests and the Mac tests are passing fine there. I myself do not have a Mac so I am afraid I can't really dig into this issue. Sorry about that! I will leave this issue open in the hopes that maybe someone will be able to figure out what is going on here.

**Comment by ZachSand** [comment]
I've found the issue. It looks like the original exception/problem got gobbled up. I added await on the scraper.py because I hunch it was having an issue making the connection.
`urls = [await self.generate_product_url(part) for part in parts]`

This lead to seeing the actual issue which was an SSL Cert issue. I didn't RTFM when installing Python 3.8 for MacOS 

> Please read the "Important Information" displayed during installation for information about SSL/TLS certificate validation and the running the "Install Certificates.command".

After running the command, I'm now able to use the API as intended. Apologies for the confusion.
